lib/engine/route.rb

```diff
@@ -4,9 +4,10 @@
 
 module Engine
   class Route
+    attr_accessor :num_halts
```

Comment:
why is this an accesor?

Reply:
Because on a brand new route that contains halts, num_halts is calculated and set by @game.compute_stops. Effectively the default num_halts for a new route is set to a number that maximizes revenue first, subsidies second.

Reply:
This is how num_halts ever becomes anything other than nil for a new route.